frontend/src/components/NodePathHistory/NodePathHistory.jsx

```diff
@@ -0,0 +1,167 @@
+  return path.join("/");
+}
+
+export const NodePathHistory = ({
+  path,
+  onNodeNameClick,
+  onPathChange,
+  isValidPath,
+}) => {
+  const [mode, setMode] = useState(MODES.VIEW);
+  const [currentText, setCurrentText] = useState("");
+  const [errorOpen, setErrorOpen] = React.useState(false);
+
+  useEffect(() => {
+    setCurrentText(convertPathToString(path));
+  }, [path]);
+
+  /**
+   * Store the current text being typed
+   */
+  const onTextChangeHandler = (event) => {
+    setCurrentText(event.target.value);
+  };
+
+  /**
+   * Triggered when the user is finished editing the path
+   * and clicks done. Notifies the parent of the change
+   * through a callback, passing in the new path as a parameter.
+   */
+  const onDoneClickHandler = () => {
+    // Remove preceding or trailing slashes and spaces
+    const newPath = currentText
+      .trim()
+      .replace(/^\/+/, "")
+      .replace(/\/+$/, "")
+      .split("/");
+
+    // Inform the parent container of the change, if there was one and
+    // given that the path is valid
+    if (currentText !== convertPathToString(path)) {
+      if (isValidPath(newPath)) {
+        onPathChange(newPath);
+      } else {
+        setErrorOpen(true);
+        clearChanges();
+      }
+    }
+  };
+
+  /**
+   * If the user clicks outside of the text field, reset
+   * the text inside of it and go back to view mode.
+   */
+  const clearChanges = () => {
+    setCurrentText(convertPathToString(path));
+    setMode(MODES.VIEW);
+  };
+
+  /**
+   * This function is called when the alert close button
+   * is clicked or there is a timeout.
+   */
+  const handleClose = (event, reason) => {
+    // Should not close if the user just clicks anywhere
+    if (reason === "clickaway") {
+      return;
+    }
+    setErrorOpen(false);
+  };
+
+  const Links = path.slice(0, -1).map((name, index) => (
+    <Button key={index} onClick={() => onNodeNameClick(name)}>
+      <Box>{name}</Box>
+    </Button>
+  ));
+
+  const BreadcrumbsView = (
+    <Breadcrumbs
+      maxItems={4}
+      itemsAfterCollapse={3}
+      itemsBeforeCollapse={0}
+      color="primary"
+      aria-label="breadcrumb"
+      separator={<NavigateNext fontSize="small" />}
+    >
+      {Links}
+
+      {/* The last element is styled differently and not rendered as a link */}
+      <Button disableRipple>
+        <Box>{path[path.length - 1]}</Box>
+      </Button>
+    </Breadcrumbs>
+  );
+
+  const EditPathView = (
+    <ClickAwayListener onClickAway={clearChanges}>
+      <TextField
+        label="Path"
+        value={currentText}
+        variant="outlined"
+        size="small"
+        onChange={onTextChangeHandler}
+        onKeyDown={(e) => {
+          // If the enter key is pressed, it is equivalent
+          // to clicking on Done
+          if (e.keyCode === 13) onDoneClickHandler();
+        }}
+        fullWidth
+      />
+    </ClickAwayListener>
+  );
+
+  const EditButton = (
```

Comment:
Wondering how easy it is to add a drop down or suggestions of some sort, like windows does in the file explorer? Because I am wondering if some users may not know that they have to have a slash between elements in the path, like this: 
![image](https://user-images.githubusercontent.com/29286974/107516695-af03c300-6ba4-11eb-9086-4ffabf8d9a4f.png)

In any case, I would suggest when the user clicks to edit the path, to not only have the name of the node but use the name of the node plus a "/" so that they have some hint as to how the feature works -> Instead of this 
![image](https://user-images.githubusercontent.com/29286974/107516819-db1f4400-6ba4-11eb-9b3e-70ff7e416a32.png) maybe have this 
![image](https://user-images.githubusercontent.com/29286974/107516862-ea9e8d00-6ba4-11eb-8a65-6693d9e98738.png)

Reply:
If we do the drop down idea, maybe that should be a separate ticket and MR tho

Reply:
Yes, let's leave the dropdown idea for now since it would require quite a bit more effort. But good suggestion about the slash, I'll add that in.